File: problems-originals/2001-2100/2075_decode-the-slanted-ciphertext/solution.cpp
class Solution {
  public:
    string decodeCiphertext(string encodedText, int rows) {
        if (encodedText.empty()) {
            return "";
        }
        int cols = static_cast<int>(encodedText.size()) / rows;
        string decoded;
        decoded.reserve(encodedText.size());
        for (int start = 0; start < cols; ++start) {
            for (int row = 0, col = start; row < rows && col < cols; ++row, ++col) {
                decoded.push_back(encodedText[row * cols + col]);
            }
        }
        while (!decoded.empty() && decoded.back() == ' ') {
            decoded.pop_back();
        }
        return decoded;
    }
};

File: problems-originals/2001-2100/2075_decode-the-slanted-ciphertext/solution.py
class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        if not encodedText:
            return ""
        cols = len(encodedText) // rows
        decoded = []
        for start in range(cols):
            row = 0
            col = start
            while row < rows and col < cols:
                decoded.append(encodedText[row * cols + col])
                row += 1
                col += 1
        return "".join(decoded).rstrip(" ")
